### PR TITLE
Remove aggregator dependency before deploying dist artifact

### LINKED ISSUE #4253: [BUG] Dependencies missing of spark-rapids v21.12.0 release jars
**Describe the bug**
We use “mvn dependency:get” command to download the rapids-4-spark jar from the OSS staging repo for the release test [spark-tests.sh#L33-L34](https://github.com/NVIDIA/spark-rapids/blob/branch-22.02/jenkins/spark-tests.sh#L33-L34).
What we want is to download the rapids-4-spark jar only, maven tries to get its dependency jars as blow, which are not available, because we will NOT release these dependencies.

We used `denpendency-reduce` before in dist/pom.xml to void some these denpendencies, but not on branch-21.12, we remove the `denpendency-reduce`  from dist/pom.xml

Related issue on branch-21.10:  #3929 #3927

```
mvn dependency:get -B -Ddest=snap -DremoteRepositories=https://oss.sonatype.org/content/repositories/staging -DgroupId=com.nvidia -DartifactId=rapids-4-spark_2.12 -Dversion=21.12.0 -s spark-rapids/jenkins/settings.xml -Dpackaging=jar

22:57:31  [INFO] ------------------------------------------------------------------------
22:57:31  [INFO] Reactor Summary for RAPIDS Accelerator for Apache Spark Root Project 21.12.0-SNAPSHOT:
22:57:31  [INFO] 
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark Root Project ... FAILURE [ 14.257 s]
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin ..... SKIPPED
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark Shuffle Plugin . SKIPPED
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Shims SKIPPED
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Spark 3.0.1 Shim SKIPPED
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark Scala UDF Plugin SKIPPED
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark Aggregator ..... SKIPPED
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark Distribution ... SKIPPED
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark UDF Examples ... SKIPPED
22:57:31  [INFO] RAPIDS Accelerator for Apache Spark Tests .......... SKIPPED
22:57:31  [INFO] rapids-4-spark-integration-tests_2.12 .............. SKIPPED
22:57:31  [INFO] rapids-4-spark-api-validation ...................... SKIPPED
22:57:31  [INFO] ------------------------------------------------------------------------
22:57:31  [INFO] BUILD FAILURE
22:57:31  [INFO] ------------------------------------------------------------------------
22:57:31  [INFO] Total time:  14.574 s
22:57:31  [INFO] Finished at: 2021-12-01T14:57:31Z
22:57:31  [INFO] ------------------------------------------------------------------------
22:57:31  [ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:2.8:get (default-cli) on project rapids-4-spark-parent: Couldn't download artifact: Missing:
22:57:31  [ERROR] ----------
22:57:31  [ERROR] 1) com.nvidia:rapids-4-spark-shuffle_2.12:jar:spark301:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR]   Try downloading the file manually from the project website.
22:57:31  [ERROR] 
22:57:31  [ERROR]   Then, install it using the command: 
22:57:31  [ERROR]       mvn install:install-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-shuffle_2.12 -Dversion=21.12.0 -Dclassifier=spark301 -Dpackaging=jar -Dfile=/path/to/file
22:57:31  [ERROR] 
22:57:31  [ERROR]   Alternatively, if you host your own repository you can deploy the file there: 
22:57:31  [ERROR]       mvn deploy:deploy-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-shuffle_2.12 -Dversion=21.12.0 -Dclassifier=spark301 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
22:57:31  [ERROR] 
22:57:31  [ERROR]   Path to dependency: 
22:57:31  [ERROR]   	1) org.apache.maven.plugins:maven-downloader-plugin:jar:1.0
22:57:31  [ERROR]   	2) com.nvidia:rapids-4-spark_2.12:jar:21.12.0
22:57:31  [ERROR]   	3) com.nvidia:rapids-4-spark-aggregator_2.12:jar:spark301:21.12.0
22:57:31  [ERROR]   	4) com.nvidia:rapids-4-spark-shuffle_2.12:jar:spark301:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR] 2) com.nvidia:rapids-4-spark-sql_2.12:jar:spark301:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR]   Try downloading the file manually from the project website.
22:57:31  [ERROR] 
22:57:31  [ERROR]   Then, install it using the command: 
22:57:31  [ERROR]       mvn install:install-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-sql_2.12 -Dversion=21.12.0 -Dclassifier=spark301 -Dpackaging=jar -Dfile=/path/to/file
22:57:31  [ERROR] 
22:57:31  [ERROR]   Alternatively, if you host your own repository you can deploy the file there: 
22:57:31  [ERROR]       mvn deploy:deploy-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-sql_2.12 -Dversion=21.12.0 -Dclassifier=spark301 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
22:57:31  [ERROR] 
22:57:31  [ERROR]   Path to dependency: 
22:57:31  [ERROR]   	1) org.apache.maven.plugins:maven-downloader-plugin:jar:1.0
22:57:31  [ERROR]   	2) com.nvidia:rapids-4-spark_2.12:jar:21.12.0
22:57:31  [ERROR]   	3) com.nvidia:rapids-4-spark-aggregator_2.12:jar:spark301:21.12.0
22:57:31  [ERROR]   	4) com.nvidia:rapids-4-spark-sql_2.12:jar:spark301:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR] 3) com.nvidia:rapids-4-spark-udf_2.12:jar:spark301:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR]   Try downloading the file manually from the project website.
22:57:31  [ERROR] 
22:57:31  [ERROR]   Then, install it using the command: 
22:57:31  [ERROR]       mvn install:install-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-udf_2.12 -Dversion=21.12.0 -Dclassifier=spark301 -Dpackaging=jar -Dfile=/path/to/file
22:57:31  [ERROR] 
22:57:31  [ERROR]   Alternatively, if you host your own repository you can deploy the file there: 
22:57:31  [ERROR]       mvn deploy:deploy-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-udf_2.12 -Dversion=21.12.0 -Dclassifier=spark301 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
22:57:31  [ERROR] 
22:57:31  [ERROR]   Path to dependency: 
22:57:31  [ERROR]   	1) org.apache.maven.plugins:maven-downloader-plugin:jar:1.0
22:57:31  [ERROR]   	2) com.nvidia:rapids-4-spark_2.12:jar:21.12.0
22:57:31  [ERROR]   	3) com.nvidia:rapids-4-spark-aggregator_2.12:jar:spark301:21.12.0
22:57:31  [ERROR]   	4) com.nvidia:rapids-4-spark-udf_2.12:jar:spark301:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR] 4) com.nvidia:rapids-4-spark-aggregator_2.12:jar:spark301:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR]   Try downloading the file manually from the project website.
22:57:31  [ERROR] 
22:57:31  [ERROR]   Then, install it using the command: 
22:57:31  [ERROR]       mvn install:install-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-aggregator_2.12 -Dversion=21.12.0 -Dclassifier=spark301 -Dpackaging=jar -Dfile=/path/to/file
22:57:31  [ERROR] 
22:57:31  [ERROR]   Alternatively, if you host your own repository you can deploy the file there: 
22:57:31  [ERROR]       mvn deploy:deploy-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-aggregator_2.12 -Dversion=21.12.0 -Dclassifier=spark301 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
22:57:31  [ERROR] 
22:57:31  [ERROR]   Path to dependency: 
22:57:31  [ERROR]   	1) org.apache.maven.plugins:maven-downloader-plugin:jar:1.0
22:57:31  [ERROR]   	2) com.nvidia:rapids-4-spark_2.12:jar:21.12.0
22:57:31  [ERROR]   	3) com.nvidia:rapids-4-spark-aggregator_2.12:jar:spark301:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR] 5) com.nvidia:rapids-4-spark-shims-spark301_2.12:jar:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR]   Try downloading the file manually from the project website.
22:57:31  [ERROR] 
22:57:31  [ERROR]   Then, install it using the command: 
22:57:31  [ERROR]       mvn install:install-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-shims-spark301_2.12 -Dversion=21.12.0 -Dpackaging=jar -Dfile=/path/to/file
22:57:31  [ERROR] 
22:57:31  [ERROR]   Alternatively, if you host your own repository you can deploy the file there: 
22:57:31  [ERROR]       mvn deploy:deploy-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-shims-spark301_2.12 -Dversion=21.12.0 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
22:57:31  [ERROR] 
22:57:31  [ERROR]   Path to dependency: 
22:57:31  [ERROR]   	1) org.apache.maven.plugins:maven-downloader-plugin:jar:1.0
22:57:31  [ERROR]   	2) com.nvidia:rapids-4-spark_2.12:jar:21.12.0
22:57:31  [ERROR]   	3) com.nvidia:rapids-4-spark-aggregator_2.12:jar:spark301:21.12.0
22:57:31  [ERROR]   	4) com.nvidia:rapids-4-spark-shims-spark301_2.12:jar:21.12.0
22:57:31  [ERROR] 
22:57:31  [ERROR] ----------
22:57:31  [ERROR] 5 required artifacts are missing.
22:57:31  [ERROR] 
22:57:31  [ERROR] for artifact: 
22:57:31  [ERROR]   org.apache.maven.plugins:maven-downloader-plugin:jar:1.0
22:57:31  [ERROR] 
22:57:31  [ERROR] from the specified remote repositories:
22:57:31  [ERROR]   apache-snapshots-repo (https://urm.nvidia.com/artifactory/sw-spark-maven, releases=true, snapshots=true),
22:57:31  [ERROR]   apache.snapshots.https (https://urm.nvidia.com/artifactory/sw-spark-maven, releases=true, snapshots=true),
22:57:31  [ERROR]   apache.snapshots (https://urm.nvidia.com/artifactory/sw-spark-maven, releases=true, snapshots=true),
22:57:31  [ERROR]   central (https://urm.nvidia.com/artifactory/sw-spark-maven, releases=true, snapshots=false),
22:57:31  [ERROR]   snapshots (https://urm.nvidia.com/artifactory/sw-spark-maven, releases=true, snapshots=true),
22:57:31  [ERROR]   snapshots-repo (https://oss.sonatype.org/content/repositories/snapshots, releases=false, snapshots=true),
22:57:31  [ERROR]   temp (https://oss.sonatype.org/content/repositories/staging, releases=true, snapshots=true)
22:57:31  [ERROR] -> [Help 1]
22:57:31  [ERROR] 
22:57:31  [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
22:57:31  [ERROR] Re-run Maven using the -X switch to enable full debug logging.
22:57:31  [ERROR] 
22:57:31  [ERROR] For more information about the errors and possible solutions, please read the following articles:
22:57:31  [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

### PR DESCRIPTION
**Pull request by gerashegalov**
- remove aggregator before publishing to maven repo
- package reduced pom with the jar in META-INF/maven
- update deploy with the new location of the reduced pom
- inconsistency of the pom installed into the local .m2 remains due to install-file bug until we refactor install as a separate step. But this inconsistency is the same as in 21.10
- missing space version-def.sh

Fixes #4253

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: jenkins/spark-tests.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ nvidia-smi
 
 ARTF_ROOT="$WORKSPACE/jars"
 MVN_GET_CMD="mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get -B \
-    -Dmaven.repo.local=$WORKSPACE/.m2 \
+    -Dmaven.repo.local=$WORKSPACE/.m2 -Dtransitive=false \
```
Comment (NvTimLiu) [inline jenkins/spark-tests.sh:26]: This fix would be good for our test script `jenkins/spark-tests.sh`, but there should be a problem on the customers' end, as they may add `rapids-4-spark_2.12` in there pom.xml without adding `transitive=false`, them they will get these dependencies error. 
Would it doable to remove the dependency tree  [dependency reduce] from `rapids-4-spark_2.12`, similar to what we did on branch-21.10? https://github.com/NVIDIA/spark-rapids/pull/3930/files#diff-2681b51a40a6a26d3a6f7f839cd810422226193ef273ea02d1b0b93c3a87cfa6R1868

Comment (jlowe) [reply jenkins/spark-tests.sh:26]: Agreed, the dependencies as published seem problematic for users who want to pull in the dist jar for building their RAPIDS accelerated UDFs or custom code that accesses ColumnarRdd.

Comment (jlowe) [reply jenkins/spark-tests.sh:26]: But in this case it's just wrong.  The dependency states that the jar needs the aggregator jar to be provided, but it contains it instead.  What's the point of advertising a bogus dependency?  It clearly causes problems for some use cases, and is misleading at best if someone examines the pom.

Comment (gerashegalov) [reply jenkins/spark-tests.sh:26]: I disagree: `provided` is the weakest scope that still serves the purpose that the aggregator dependency tree is required for building the dist jar (which it is, especially given that some branches like spark312db are built remotely). The next weakest dependency scope is `runtime` stating that the dependency will be needed only at the execution phase, and using that might be misleading. 

> What's the point of advertising a bogus dependency?
disagree with the characterization as bogus because the aggregator is required for building dist package. According to the transitive dependency rules: if the `application` depends with any scope on `dist` and `dist` depends with `provided` on `aggregator`, the dependency subtree rooted in `aggregator` is specifically not advertised.

Comment (jlowe) [reply jenkins/spark-tests.sh:26]: IMO this violates the spirit of the shading we're doing in two ways.

1. Most shaded jars do not advertise the dependencies they used during builds when they pull in those artifacts.  That's why there's a dependency-reduced pom built by default by the shade plugin.  What we're doing here is effectively a form of shading, so I do not understand why we would want to advertise the details of a dependency that is irrelevant once the artifact is published.  What's the use-case where having this dependency in the published artifact is desired?  Maybe I'm missing that.

2. I do not think we should reference unpublished dependencies in the poms of our published artifacts, even if they are marked provided. The user does not need to provide it to use the artifact, and the fact that this is a build-time only dependency is not clearly conveyed by the pom as-is.

Bottom line is that this is breaking some workflows as it is now.  Sure, we control this particular workflow and can update it as a workaround, but do we really want to explain this all over again to a user that is doing something similar?  I think it's a bit difficult to argue why we need to list a dependency in this pom as provided when it's not actually needed and then didn't bother to publish a dependency-reduced pom to avoid such problems.

Comment (gerashegalov) [reply jenkins/spark-tests.sh:26]: We have tried manual install-file with dependency-reduced pom https://github.com/NVIDIA/spark-rapids/pull/4043/commits/372bc1387f822f3b9048d866a5be7e3b25770740#diff-2681b51a40a6a26d3a6f7f839cd810422226193ef273ea02d1b0b93c3a87cfa6R375 

The issue with that is a bug in the install-file mojo when executed inside a multi-module build it's picking a random (based  on the build order of the current maven session) instead of the current pom. I question whether it's worth solving a simple problem affecting this edge case because even a small change is more disruptive https://github.com/NVIDIA/spark-rapids/pull/4043#issuecomment-965766850 than the problem it addresses.

**Comment by pxLi** [comment]
Would this be better to target `branch-21.12`?

File: jenkins/spark-tests.sh (hunk 1 above)
Comment (gerashegalov) [reply jenkins/spark-tests.sh]: Updated the PR based on the offline meeting with @jlowe

**Comment by NvTimLiu** [comment]
Got build error with this PR
```
19:45:30  [ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.0.0:run (create-parallel-world) on project rapids-4-spark_2.12: An Ant BuildException has occured: The following error occurred while executing this line:
19:45:30  [ERROR] /home/jenkins/agent/workspace/jenkins-tim-rapids-release-7/dist/maven-antrun/build-parallel-worlds.xml:125: exec returned: 255
19:45:30  [ERROR] around Ant part ...<ant antfile="/home/jenkins/agent/workspace/jenkins-tim-rapids-release-7/dist/maven-antrun/build-parallel-worlds.xml" target="build-parallel-worlds" />... @ 9:156 in /home/jenkins/agent/workspace/jenkins-tim-rapids-release-7/dist/target/antrun/build-main.xml
19:45:30  [ERROR] -> [Help 1]
19:45:30  [ERROR] 
19:45:30  [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
19:45:30  [ERROR] Re-run Maven using the -X switch to enable full debug logging.
19:45:30  [ERROR] 
19:45:30  [ERROR] For more information about the errors and possible solutions, please read the following articles:
19:45:30  [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
19:45:30  [ERROR] 
19:45:30  [ERROR] After correcting the problems, you can resume the build with the command
19:45:30  [ERROR]   mvn <goals> -rf :rapids-4-spark_2.12
```

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
The error is from a line number preceding the change in the PR, probably unrelated but checking.

**Comment by gerashegalov** [comment]
Inspecting the CI workspace points to the Databricks-vs-Other-Shims split brain problem.  
```
$ grep ExplainPlanImpl.class tmp-sha1*.txt
d256e7b9639b757d13e650e96eaee1850ec05f13 *./parallel-world/spark301/com/nvidia/spark/rapids/ExplainPlanImpl.class
22aee3f24a0fa81ce3e26acc42641b520cfa1e03 *./parallel-world/spark301db/com/nvidia/spark/rapids/ExplainPlanImpl.class
d256e7b9639b757d13e650e96eaee1850ec05f13 *./parallel-world/spark302/com/nvidia/spark/rapids/ExplainPlanImpl.class
d256e7b9639b757d13e650e96eaee1850ec05f13 *./parallel-world/spark303/com/nvidia/spark/rapids/ExplainPlanImpl.class
d256e7b9639b757d13e650e96eaee1850ec05f13 *./parallel-world/spark311/com/nvidia/spark/rapids/ExplainPlanImpl.class
d256e7b9639b757d13e650e96eaee1850ec05f13 *./parallel-world/spark311cdh/com/nvidia/spark/rapids/ExplainPlanImpl.class
d256e7b9639b757d13e650e96eaee1850ec05f13 *./parallel-world/spark312/com/nvidia/spark/rapids/ExplainPlanImpl.class
22aee3f24a0fa81ce3e26acc42641b520cfa1e03 *./parallel-world/spark312db/com/nvidia/spark/rapids/ExplainPlanImpl.class
d256e7b9639b757d13e650e96eaee1850ec05f13 *./parallel-world/spark320/com/nvidia/spark/rapids/ExplainPlanImpl.class

$ grep ExplainPlanImpl.class tmp-sha1*.txt | sort -u -k1,1
22aee3f24a0fa81ce3e26acc42641b520cfa1e03 *./parallel-world/spark301db/com/nvidia/spark/rapids/ExplainPlanImpl.class
d256e7b9639b757d13e650e96eaee1850ec05f13 *./parallel-world/spark301/com/nvidia/spark/rapids/ExplainPlanImpl.class
```

The jars being combined were built at different commits, c8dd50132676feeccd7e4908e8f7937b97c62c87 and 6fd125d70d204a5402a0853a367d9f4f99ed8a22 respectively
```
$ unzip -p deps/rapids-4-spark-aggregator_2.12-21.12.0-spark312db.jar rapids4spark-version-info.properties
version=21.12.0
cudf_version=21.12.0-SNAPSHOT
user=ubuntu
revision=c8dd50132676feeccd7e4908e8f7937b97c62c87
branch=HEAD
date=2021-11-30T11:46:24Z
url=https://github.com/NvTimLiu/spark-rapids.git
```
vs
```
$ unzip -p deps/rapids-4-spark-aggregator_2.12-21.12.0-spark320.jar rapids4spark-version-info.propertiesversion=21.12.0
cudf_version=21.12.0-SNAPSHOT
user=
revision=6fd125d70d204a5402a0853a367d9f4f99ed8a22
branch=HEAD
date=2021-12-03T11:01:38Z
url=https://github.com/NvTimLiu/spark-rapids.git
```

**Comment by NvTimLiu** [comment]
> 

Oh, I see, we need to build out DB shims before the spark-rapids release. 

If this 'c8dd501 and 6fd125d respectively' the reason for the error `build-parallel-worlds.xml:125: exec returned: 255` ?

Let me update the DB shims using the same commit Id with the Other-Shims.

**Comment by gerashegalov** [comment]
> Oh, I see, we need to build out DB shims before the spark-rapids release.
> 
> Is this diff `c8dd501 and 6fd125d respectively` the reason for the error `build-parallel-worlds.xml:125: exec returned: 255` ?

yes, it's because of the commit mismatch because I verified that the the shas match and the build succeeds once the nightly artifacts are caught up. Somehow we had an an inconsistent set of artifacts in the internal maven repo, which I could reproduce by skipping the real build but just building `-pl dist` for `noSnapshotsWithDatabricks` without the dependency modules (`-am`). Some hours later, next nightly build finished and the artifacts became consistent again.

I think there is a simple solution to the problem that totally prevents this from happening. When databricks is involved, all artifacts (shims) should be built on databricks nodes. Then the build pipeline can fan out into testing stages and only databricks-related tests are run on databricks, where as the rest runs on non-databricks fleet. We should not fan-out for artifact creation as we do today.

I'll also file a bug to surface the error message from binary-dedupe.sh to maven better.

**Comment by NvTimLiu** [comment]
Updated DB shims using the same commit Id with the nightly ones, build PASS, test PASS.

Build:  `tim-databricks-release-urm #1
rapids-release #90`


Test:  REL_it-A30-AQE/19/consoleFull

```
11:39:26  + mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get -B -Dmaven.repo.local=/home/jenkins/agent/workspace/REL_it-A30-AQE/.m2 -s jenkins/settings.xml -P mirror-apache-to-urm -Ddest=/home/jenkins/agent/workspace/REL_it-A30-AQE/jars -DremoteRepositories=https://oss.sonatype.org/content/repositories/staging -DgroupId=com.nvidia -DartifactId=rapids-4-spark_2.12 -Dversion=21.12.0
11:39:27  [INFO] Scanning for projects...
11:39:28  [INFO] ------------------------------------------------------------------------
11:39:28  [INFO] Reactor Build Order:
11:39:28  [INFO] 
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark Root Project                   [pom]
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin                     [jar]
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark Shuffle Plugin                 [jar]
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Shims               [pom]
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Spark 3.0.1 Shim    [jar]
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark Scala UDF Plugin               [jar]
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark Aggregator                     [jar]
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark Distribution                   [jar]
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark UDF Examples                   [jar]
11:39:28  [INFO] RAPIDS Accelerator for Apache Spark Tests                          [jar]
11:39:28  [INFO] rapids-4-spark-integration-tests_2.12                              [jar]
11:39:28  [INFO] rapids-4-spark-api-validation                                      [jar]
11:39:28  [INFO] 
11:39:28  [INFO] ------------------< com.nvidia:rapids-4-spark-parent >------------------
11:39:28  [INFO] Building RAPIDS Accelerator for Apache Spark Root Project 21.12.0-SNAPSHOT [1/12]
11:39:28  [INFO] --------------------------------[ pom ]---------------------------------
11:39:28  [INFO] 
11:39:28  [INFO] --- maven-dependency-plugin:2.8:get (default-cli) @ rapids-4-spark-parent ---
11:39:29  [INFO] Resolving com.nvidia:rapids-4-spark_2.12:jar:21.12.0 with transitive dependencies
11:39:29  [INFO] Downloading from apache-snapshots-repo: https://urm.nvidia.com/artifactory/sw-spark-maven/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.pom
11:39:31  [INFO] Downloading from temp: https://oss.sonatype.org/content/repositories/staging/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.pom
11:39:31  [INFO] Downloaded from temp: https://oss.sonatype.org/content/repositories/staging/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.pom (16 kB at 27 kB/s)
11:39:31  [INFO] Downloading from apache-snapshots-repo: https://urm.nvidia.com/artifactory/sw-spark-maven/com/nvidia/rapids-4-spark-parent/21.12.0/rapids-4-spark-parent-21.12.0.pom
11:39:32  [INFO] Downloading from temp: https://oss.sonatype.org/content/repositories/staging/com/nvidia/rapids-4-spark-parent/21.12.0/rapids-4-spark-parent-21.12.0.pom
11:39:32  [INFO] Downloaded from temp: https://oss.sonatype.org/content/repositories/staging/com/nvidia/rapids-4-spark-parent/21.12.0/rapids-4-spark-parent-21.12.0.pom (65 kB at 222 kB/s)
11:39:32  [INFO] Downloading from apache.snapshots.https: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml
11:39:32  [INFO] Downloading from apache-snapshots-repo: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml
11:39:32  [INFO] Downloading from snapshots: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml
11:39:32  [INFO] Downloading from apache.snapshots: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml
11:39:32  [INFO] Downloaded from apache.snapshots.https: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml (1.6 kB at 44 kB/s)
11:39:32  [INFO] Downloaded from apache-snapshots-repo: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml (1.6 kB at 44 kB/s)
11:39:32  [INFO] Downloaded from apache.snapshots: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml (1.6 kB at 43 kB/s)
11:39:32  [INFO] Downloading from temp: https://oss.sonatype.org/content/repositories/staging/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml
11:39:32  [INFO] Downloaded from snapshots: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml (1.6 kB at 41 kB/s)
11:39:32  [INFO] Downloading from snapshots-repo: https://oss.sonatype.org/content/repositories/snapshots/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml
11:39:32  [INFO] Downloaded from snapshots-repo: https://oss.sonatype.org/content/repositories/snapshots/ai/rapids/cudf/21.12.0-SNAPSHOT/maven-metadata.xml (1.4 kB at 8.7 kB/s)
11:39:32  [INFO] Downloading from apache-snapshots-repo: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/cudf-21.12.0-20211204.082229-68.pom
11:39:32  [INFO] Downloaded from apache-snapshots-repo: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/cudf-21.12.0-20211204.082229-68.pom (23 kB at 830 kB/s)
11:39:32  [INFO] Downloading from apache-snapshots-repo: https://urm.nvidia.com/artifactory/sw-spark-maven/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.jar
11:39:32  [INFO] Downloading from apache-snapshots-repo: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/cudf-21.12.0-20211204.082229-68-cuda11.jar
11:39:33  [INFO] Downloading from temp: https://oss.sonatype.org/content/repositories/staging/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.jar
11:39:35  [INFO] Downloaded from temp: https://oss.sonatype.org/content/repositories/staging/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.jar (34 MB at 17 MB/s)
11:39:39  [INFO] Downloaded from apache-snapshots-repo: https://urm.nvidia.com/artifactory/sw-spark-maven/ai/rapids/cudf/21.12.0-SNAPSHOT/cudf-21.12.0-20211204.082229-68-cuda11.jar (329 MB at 56 MB/s)
11:39:39  [WARNING] destination/dest parameter is deprecated: it will disappear in future version.
11:39:39  [INFO] Copying /home/jenkins/agent/workspace/REL_it-A30-AQE/.m2/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.jar to /home/jenkins/agent/workspace/REL_it-A30-AQE/jars
11:39:39  [WARNING] Notice transitive dependencies won't be copied.
11:39:39  [INFO] ------------------------------------------------------------------------
11:39:39  [INFO] Reactor Summary for RAPIDS Accelerator for Apache Spark Root Project 21.12.0-SNAPSHOT:
11:39:39  [INFO] 
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark Root Project ... SUCCESS [ 10.762 s]
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin ..... SKIPPED
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark Shuffle Plugin . SKIPPED
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Shims SKIPPED
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Spark 3.0.1 Shim SKIPPED
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark Scala UDF Plugin SKIPPED
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark Aggregator ..... SKIPPED
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark Distribution ... SKIPPED
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark UDF Examples ... SKIPPED
11:39:39  [INFO] RAPIDS Accelerator for Apache Spark Tests .......... SKIPPED
11:39:39  [INFO] rapids-4-spark-integration-tests_2.12 .............. SKIPPED
11:39:39  [INFO] rapids-4-spark-api-validation ...................... SKIPPED
11:39:39  [INFO] ------------------------------------------------------------------------
11:39:39  [INFO] BUILD SUCCESS
11:39:39  [INFO] ------------------------------------------------------------------------
11:39:39  [INFO] Total time:  11.035 s
11:39:39  [INFO] Finished at: 2021-12-05T03:39:38Z
11:39:39  [INFO] ------------------------------------------------------------------------
```

**Review by NvTimLiu — APPROVED** [review]

**Comment by NvTimLiu** [comment]
@gerashegalov  Could you please help to check if we're good to merge?

**Comment by GaryShen2008** [comment]
Merge it since test passed.
